backend/core/src/providers/heroic_gog_provider.rs

```diff
@@ -0,0 +1,269 @@
+struct Extra {
+	about: About,
+	reqs: Vec<String>,
+	genres: Vec<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct About {
+	description: String,
+	#[serde(rename(deserialize = "shortDescription"))]
+	short_description: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Timestamp {
+	games: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Root {
+	games: Vec<Game>,
+}
+
+fn get_detected_games() -> Result<Vec<Game>, io::Error> {
+	let dirs = BaseDirs::new().expect("Failed to get user directories");
+	let config_dir = dirs.config_dir();
+	let file_content =
+		read_to_string(Path::new(&config_dir).join("heroic/store_cache/gog_library.json"))?;
+	let installed_games = read_installed_games()?;
+
+	Ok(serde_json::from_str::<Root>(file_content.as_str())?
+		.games
+		.into_iter()
+		// gog-redist is not a game but it shows up in the library
+		.filter(|game| game.app_name != "gog-redist".to_string())
+		.map(|mut game| {
+			if installed_games.contains(&game.app_name) {
+				// is_installed props from the library are not reliable
+				game.is_installed = true;
+			}
+			game
+		})
+		.collect())
+}
+
+fn read_installed_games() -> Result<Vec<String>, io::Error> {
+	let dirs = BaseDirs::new().expect("Failed to get user directories");
+	let config_dir = dirs.config_dir();
+	let file_content =
+		read_to_string(Path::new(&config_dir).join("heroic/gog_store/installed.json"))?;
+
+	Ok(
+		serde_json::from_str::<RootInstalled>(file_content.as_str())?
+			.installed
+			.into_iter()
+			.map(|game| game.app_name)
+			.collect(),
+	)
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct PlayTask {
+	category: String,
+	is_primary: Option<bool>,
+	languages: Vec<String>,
+	name: String,
+	path: Option<String>,
+	type_: String,
+	link: Option<String>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct GogGame {
+	build_id: String,
+	client_id: String,
+	game_id: String,
+	language: String,
+	languages: Vec<String>,
+	name: String,
+	play_tasks: Vec<PlayTask>,
+	root_game_id: String,
+	version: i32,
+}
+
+fn read_info_file(path: &Path, app_id: &str) -> Result<GogGame, String> {
+	let file_path = path.join(format!("goggame-{}.info", app_id));
+
+	match fs::read_to_string(file_path) {
+		Ok(json_string) => match serde_json::from_str(&json_string) {
+			Ok(json_value) => Ok(json_value),
+			Err(err) => Err(format!("Error parsing JSON: {}", err)),
+		},
+		Err(err) => Err(format!("Error reading file: {}", err)),
+	}
+}
+
+#[derive(Clone)]
+pub struct HeroicGog {}
+
+impl HeroicGog {
+	fn get_installed_game(entry: &Game) -> Option<InstalledGame> {
+		let dirs = BaseDirs::new().expect("Failed to get user directories");
+		let home_dir = dirs.home_dir();
+		let game_path = Path::new(&home_dir)
+			.join("Games/Heroic")
+			.join(&entry.folder_name.clone().unwrap());
+		let infos = read_info_file(game_path.as_path(), &entry.app_name).unwrap();
+
```

Comment:
I set up clippy rules across Rai Pal to force me to properly handle all errors, so unwraps aren't allowed. Same goes for anything else that can easily trigger a panic, including assert or explicit panic calls.

Reply:
You should be getting these clippy warnings but maybe it requires configuring the IDE to make them show, not sure

Reply:
Alright, gonna patch these. I might configure clippy as a pre-commit hook.